pkg/lint/rules/crds_test.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rules
+
+import (
+	"strings"
+	"testing"
+
+	"helm.sh/helm/v4/pkg/lint/support"
+)
+
+const crdsTestBaseDir = "./testdata/withcrd"
+const invalidCrdsDir = "./testdata/invalidcrdsdir"
+
+func TestCrdsDir(t *testing.T) {
+	linter := support.Linter{ChartDir: crdsTestBaseDir}
+	Crds(&linter)
+	res := linter.Messages
+
+	if len(res) > 0 {
+		t.Fatalf("Expected no errors, got %d, %v", len(res), res)
+	}
```

Comment:
we make heavy usage of https://github.com/stretchr/testify in Helm already, this could be written as:
```
assert.Emptyf(t, res, "Expected no errors, got %v", res)
```